tokio-util/src/lib.rs

```diff
@@ -69,7 +69,45 @@ mod util {
     use std::pin::Pin;
     use std::task::{Context, Poll};
 
-    pub(crate) fn poll_read_buf<T: AsyncRead>(
+    /// Try to read data from an `AsyncRead` into an implementer of the [`Buf`] trait.
+    ///
+    /// [`Buf`]: bytes::Buf
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use bytes::{Bytes, BytesMut};
+    /// use tokio::stream;
+    /// use tokio::io::Result;
+    /// use tokio_util::io::{StreamReader, poll_read_buf};
+    /// use futures::future::poll_fn;
+    /// use std::pin::Pin;
+    /// # #[tokio::main]
+    /// # async fn main() -> std::io::Result<()> {
+    ///
+    /// // Create a reader from an iterator. This particular reader will always be
+    /// // ready.
+    /// let mut read = StreamReader::new(stream::iter(vec![Result::Ok(Bytes::from_static(&[0, 1, 2, 3]))]));
+    ///
+    /// let mut buf = BytesMut::new();
+    /// let mut reads = 0;
+    ///
+    /// loop {
+    ///     reads += 1;
+    ///     let n = poll_fn(|cx| poll_read_buf(cx, Pin::new(&mut read), &mut buf)).await?;
+    ///
+    ///     if n == 0 {
+    ///         break;
+    ///     }
+    /// }
+    ///
+    /// // one or more reads might be necessary.
+    /// assert!(reads >= 1);
+    /// assert_eq!(&buf[..], &[0, 1, 2, 3]);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn poll_read_buf<T: AsyncRead>(
         cx: &mut Context<'_>,
         io: Pin<&mut T>,
         buf: &mut impl BufMut,
```

Comment:
I think I would prefer something like this. What do you think?
```suggestion
    pub fn poll_read_buf<T: AsyncRead, B: BufMut>(
        io: Pin<&mut T>,
        cx: &mut Context<'_>,
        buf: &mut B,
```

Reply:
Sure, I stupidly applied your suggestion from the previous PR where this was in an impl on `AsyncReadExt`. Personally, I think it makes more sense to put the `cx` argument last, but in the end it doesn't matter that much.